src/unsafe_string.rs

```diff
@@ -18,7 +20,7 @@ use super::value::{IValue, TypeTag};
 #[repr(C)]
 #[repr(align(4))]
 struct Header {
-    rc: usize,
+    rc: AtomicUsize,
```

Comment:
So now we will use the atomic integers also for the single-threaded environments, but they introduce extra costs compared to the non-atomic integers. Should we, perhaps, introduce something like a `MaybeAtomic`, but patched to enable the atomicity at runtime-level instead? In the end, we just need a register to store the data and either perform the atomic instructions on it or not. The current `MaybeAtomic` crate chooses this at compile-time, but we can make the same decisions at runtime.

Have you already run any benchmarks with it locally?

Reply:
I trigger the RedisJSON performance CI, base on the results we can decide.

Reply:
@filipecosta90 confirm no regression is detected.